### ISSUE TITLE
Rancher is frozen when User click on 'Auth Provider' button once it is already connected to an Auth

### ISSUE THREAD

**Issue by dasarinaidu** [rancher/dashboard#7818]
**Setup**
- Rancher version: v2.7-c81b8b45136f3b5d26d41dca9ef0ddcabd2e026c-head
- Browser type & version: Chrome
- Modes : Dark and Auto

**Describe the bug**
Rancher is frozen when User click on 'Auth Provider' button once it is already loaded. Window is frozen and it says loading but it never loads. User has to refresh the browser to perform other operations.

**To Reproduce**
1. Login to Rancher
2. Connect to one of the Auth Provider (AZUR)
3. Once it is connected 
4. Click on the 'Auth Provider' button again
5. Check for the system behavior

**Result**
1. Rancher window is grayed out and it says the page is loading but it never loads
2. No other buttons are working to perform any operation
3. User has to refresh the browser or hard refresh with keyboard to see rancher up.

**Expected Result**
Once user click on 'Auth Provider' button it should be on the same screen and it should not disable the screen. User should able to click on other buttons without a browser refresh.

**Note :** I m not seeing Loading page on any other screen navigations OR I am not seeing Loading Page after a page refresh

**Screenshots**

**Additional context**
Seeing the same issue on v2.6.9 also.

**Comment by gaktive** [comment]
We should confirm the behaviour with another auth provider (e.g, Github) and confirm if the behaviour is happening tied to the mechanism.

@richard-cox already did a repro with Github that brings up a loading indicator (full screen Loading... with no action) which seems to point to a problem with a redirect with the side nav (Auth Provider).

**Comment by nwmac** [comment]
Note this can also be reproduced with GitHub auth

**Comment by dasarinaidu** [comment]
I have validated this issue on `v2.7-6316baf1697cca085e073ccd5e9e7a3bc542d0e5-head` . This issue got fixed and I m closing this card.

**Validation details as below:**
Rancher Version : v2.7-6316baf1697cca085e073ccd5e9e7a3bc542d0e5-head
Browsers : Chrome and Firefox
Users : Admin and Standard Users
Modes : Auto and Dark

Test Cases:

1. Admin user
2. Test User

Steps to Reproduce:

1. Login to Rancher
2. Connect to one of the Auth Provider (OpenLDAP)
3. Once it is connected
4. Click on the 'Auth Provider' button again
5. Check for the system behavior

Actual: 
1. Rancher is not frozen and navigating to appropriate screen.
